python_modules/libraries/dagster-dg-cli/dagster_dg_cli/cli/api/API_IMPLEMENTATION_PLAN.md

```diff
@@ -0,0 +1,413 @@
+# Dagster Plus API CLI Implementation Plan
+
+## Standard Verbs Pattern
+
+- **list** - List all resources with filtering/pagination
+- **view <id>** - Show detailed information about specific resource
+- **create** - Create new resource (where applicable)
+- **update <id>** - Update existing resource (where applicable)
+- **delete <id>** - Remove resource (where applicable)
+
+---
+
+## **Tier 1: Core Infrastructure**
+
+### 1. **deployment** ✅ _Already implemented_
+
+```bash
+# Implemented
+dg api deployment list [--json]
+
+# Future verbs
+dg api deployment view <name> [--json]
+dg api deployment create --name <name> --type <serverless|hybrid>
+dg api deployment update <name> [--settings <file>]
+dg api deployment delete <name>
+```
+
+**GraphQL**: `fullDeployments`, `currentDeployment`, `deploymentByName`
+
+### 2. **secret**
+
+```bash
+dg api secret list [--location <name>] [--scope <deployment|organization>] [--json]
+dg api secret view <name> [--location <name>] [--json]
+dg api secret create --name <name> --value <value> [--location <name>] [--scope <scope>]
+dg api secret update <name> --value <value> [--location <name>]
+dg api secret delete <name> [--location <name>]
+```
+
+**GraphQL**: `secretsOrError`
+
+### 3. **agent**
+
+```bash
+dg api agent list [--json]
+dg api agent view <id> [--json]
+# Agents are typically managed via configuration, not CRUD operations
+```
+
+**GraphQL**: `agents`
+
+### 4. **run**
+
+```bash
+dg api run list [--status <status>] [--pipeline <name>] [--limit <n>] [--json]
+dg api run view <run-id> [--json]
+dg api run create --pipeline <name> [--config <file>] [--tags <key=value>]
```

Comment:
the forbidden noun ('pipeline') is still popping in here and there. This should be job right?